src/api-client.js

```diff
@@ -0,0 +1,143 @@
+import fetch from 'node-fetch';
+import httpStatus from 'http-status';
+import {URL} from 'url';
+import {createLogger, logError, generateAuthorizationHeader} from './utils';
+import ApiError from './error';
+import {MarcRecord} from '@natlibfi/marc-record';
+
+// Change to true when working
+MarcRecord.setValidationOptions({fields: false, subfields: false, subfieldValues: false});
+
+export function createApiClient({restApiUrl, restApiUsername, restApiPassword, cataloger = false, userAgent = 'Melinda commons API client / Javascript'}) {
+	const logger = createLogger();
+	const Authorization = generateAuthorizationHeader(restApiUsername, restApiPassword);
+
+	const defaultParamsBulk = cataloger ? {pCatalogerIn: cataloger} : {};
+	const defaultParamsPrio = cataloger ? {cataloger} : {};
+
+	return {
+		getRecord, postPrio, postBulk, getMetadata,
+		getStatus, deleteBulk
+	};
```

Comment:
Expose only the stuff the API clients need:
- create
- update
- read
- createBulk
- readBulk